core/src/main/kotlin/com/rudderstack/sdk/kotlin/core/internals/policies/backoff/ExponentialBackOffPolicy.kt

```diff
@@ -0,0 +1,55 @@
+package com.rudderstack.sdk.kotlin.core.internals.policies.backoff
+
+import com.rudderstack.sdk.kotlin.core.internals.utils.InternalRudderApi
+import java.security.SecureRandom
+import kotlin.math.pow
+
+internal const val MIN_INTERVAL = 10L
+internal const val MAX_INTERVAL = 60_000L
+internal const val DEFAULT_INTERVAL = 3000L
+
+internal const val MIN_BASE = 1.1
+internal const val MAX_BASE = 5.0
+internal const val DEFAULT_BASE = 2.0
+
+/**
+ * This class implements an exponential backoff strategy with jitter for handling retries.
+ * It allows for configurable interval and base for the exponential calculation.
+ * The next delay is calculated using the formula: delay = interval * base^attempt.
+ * The delay is then adjusted with a random jitter to avoid synchronized retries.
+ */
+@InternalRudderApi
+class ExponentialBackOffPolicy(
+    private var intervalInMillis: Long = DEFAULT_INTERVAL,
```

Comment:
Rename `intervalInMillis` to `minDelayInSec` or `initialDelayInSecs`. Refer to our internal doc for reference.

`intervalInMillis` doesn't represent the exact task of this variable. 

This variable is needed to ensure that the first delay happens after a fixed interval. To better reflect the use case, we should rename it to something that helps us understand this better.

```suggestion
    private var minDelayInSec: Int = DEFAULT_DELAY_IN_SEC,
```